src/syscall_handler.c

```diff
@@ -86,6 +87,27 @@ int sys_run( const char *path )
     for(i=0;i<p->window_count;i++) {
         p->windows[i]->count++;
     }
+
```

Comment:
Move most of this into a new function in the process module to setup the stack for a new process.
We want these sys_X routines to be pretty simple wrappers around kernel functionality in other modules.